doc/design.md

```diff
@@ -0,0 +1,564 @@
+# Secure FHIR Data Access Design Doc
+
+In the context of healthcare data systems in resource constrained environments
+
+GitHub repository: https://github.com/google/fhir-access-proxy
+
+## Summary
+
+### The problem
+
+We want to create an access-control _gateway_ that is deployable in front of any
+generic [FHIR](https://www.hl7.org/fhir/) server. The job of this gateway is to
+let authenticated users access only what they are authorized to. In most cases,
+this gateway acts as a reverse proxy. We want this proxy server to be scalable,
+independent of the FHIR store implementation, and deployable both on-premise and
+in any cloud. We also want the access rules to be customizable and fine-grained.
+For more context, please see the [Introduction](#introduction) section.
+
+### Summary of the approach
+
+Here is a summary of proposals and decisions in this design document, for the
+first version (V1) of the FHIR gateway:
+
+![Architecture](summary.png)
```

Comment:
Is it possible to make this and all other architecture images smaller, for example limit the width to 50%?